sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -247,6 +291,23 @@ public void AddTag(string name, object value)
                     {
                         linkCollection.Add(link);
                     }
+#else
+                    ActivityTagsCollection linkTagsCollection = new();
+                    foreach (var tag in activity.Tags)
+                    {
+                        linkTagsCollection.Add(tag.Key, tag.Value!);
+                    }
+
+                    try
+                    {
+                        var context = ActivityContext.Parse(activity.ParentId!, activity.TraceStateString);
+                        var link = new ActivityLink(context, linkTagsCollection);
+                        linkCollection.Add(link);
+                    }
+                    catch (Exception)
+                    {
+                    }
```

Comment:
I wrote it this way to preserve the current behavior with [reflection](url):
```
Expression.TryCatch(
          Expression.Convert(Expression.New(ctor,
                    Expression.Call(parseMethod, traceparentParameter, tracestateParameter),
          Expression.Convert(tagsParameter, ActivityTagsCollectionType)), typeof(object)),
Expression.Catch(typeof(Exception), Expression.Default(typeof(object)))),
```

The exception being caught is an "invalid trace parent" ArgumentException, based on this [test](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/tests/ClientDiagnosticsTests.Net50.cs#L380) my understanding is that it's just supposed to be ignored.

I do think it could be helpful to add some logging since the link isn't added when it fails, but after some investigation I'm actually not sure what logging would look like in this scenario. The only other logging is for scope level events (start, stop, scope failed).

Reply:
+1 on logging, to give someone a chance of finding the issue if it's ever hit for an unexpected exception type.  Jeff Richter used to like to make the point that catching a top-level exception will catch even things like OutOfMemoryExceptions, which may not be thrown by our code even.  In general, we try to avoid this, but if it is there to preserve existing behavior, I don't think we can back it out.

Reply:
We could consider using TryParse instead, and then we won't need to catch anything. I'm not sure if there was some reflection related reason for using Parse to begin with (I imagine it is easier to use Parse than trying to extract the out param in TryParse)

Reply:
@JoshLove-msft that makes sense, I wanted to preserve the behavior as much as possible to not break anything, but this can definitely be simplified since we are no longer using reflection.